datafusion/expr/src/expr_schema.rs

```diff
@@ -87,14 +91,27 @@ impl ExprSchemable for Expr {
                     .iter()
                     .map(|e| e.get_type(schema))
                     .collect::<Result<Vec<_>>>()?;
-                Ok((fun.return_type)(&data_types)?.as_ref().clone())
+
+                // TODO not convert to DataType
+                let data_types = data_types
+                    .into_iter()
+                    .map(|e| e.physical_type())
+                    .collect::<Vec<_>>();
+
+                Ok((fun.return_type)(&data_types)?.as_ref().clone().into())
```

Comment:
TODO: Functions signatures for `udf/udaf` use `TypeSignature` instead of the arrow `DataType`